React InstantSearch Hooks/conditional-debouncing/src/App.js

```diff
@@ -0,0 +1,50 @@
+import algoliasearch from 'algoliasearch/lite';
+import {
+  Highlight,
+  Hits,
+  InstantSearch,
+  SearchBox,
+} from 'react-instantsearch-hooks-web';
+
+import 'instantsearch.css/themes/algolia.css';
+import './App.css';
+
+const searchClient = algoliasearch(
+  'latency',
+  '6be0576ff61c053d5f9a3225e2a90f76'
+);
+
+let timerId = undefined;
+
+function App() {
+  return (
+    <div className="container">
+      <InstantSearch indexName="instant_search" searchClient={searchClient}>
+        <SearchBox
+          queryHook={(query, search) => {
+            const timeout = navigator.connection?.effectiveType?.includes('2g')
+              ? 400
+              : 0;
```

Comment:
```suggestion
            const effectiveType = navigator.connection?.effectiveType;
            const rtt = navigator.connection?.rtt;
            const saveData = navigator.connection?.saveData;

           const timeout = effectiveType?.includes('2g') || rtt > 500 || saveData
             ? 400
             : 0;
```

Should we also base the condition on the [`navigator.connection.rtt`](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/rtt) and [`navigator.connection.saveData`](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) values?

Reply:
`effectiveType` already computes the `rtt` so we don't need to include it.

`saveData` can definitely be interesting if we know we're on a cellular network, or tethering to one. There's the `type` attribute but it's only available on Chrome OS so I'm not sure it's worth the hassle.

Reply:
why do you think effectiveType is better to use than a number? I'd prefer rtt or downlink too

Reply:
As mentioned on the related guide, `effectiveType` is already a combination of `rtt` + `downlink`. [See MDN](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType):
>This value is determined using a combination of recently observed round-trip time and downlink values.

Using `rtt` and `downlink` basically means recomputing our own `effectiveType` with our own arbitrary values. I think using `effectiveType` already does this job for us and is simpler for a code sample.